### DISCUSSION TITLE
using excalidraw on an "e-paper tablet" like remarkable2 or Onyx Boox Note Air 2

### DISCUSSION THREAD

**Discussion by thfrei** [excalidraw/excalidraw#5199] [Q&A]
Dear community

I was wondering if any of you have experience in using excalidraw and drawing with a stylus on any e-paper tablet, specifically one of the below. I guess you would run chrome on the tablet, open excalidraw.com, and then draw with the stylus.
- ~~remarkable 2~~ (so, this is not possible (duh), since it doesn't have a browser)
- Onyx Boox Note Air 2
- Onyx Boox Note 5
- or any older version or other tablet?

I am curious about how the hand-writing experience is:
- What is the lag? 
- How are the lines?
- How is the lag compared to using the "default" hand-writing app? (performance of chrome-excalidraw renderer)
- How is panning, zooming working (touch / multi-touch)?
- How do the colors appear?
- Update: Maybe a banchmark with this library can be useful? https://www.tldraw.com/

If you have an e-paper tablet, it would be great, if you could maybe add a video and have a test?

Thank you and best regards
Thomas

PS: I developed an integration for the note-taking app trilium, and therefore I am wondering, if the performance would be enough to use it via chrome browser on the tablet

**Comment by dwelle** [comment]
I don't have an e-ink tablet so can't test. I assume you're considering buying one and want to make sure Excalidraw works on it first? Would love to know as well (personally I can't imagine what the experience would be like!).

**Reply by thfrei** [reply]
Totally. I am considering buying one, and I wonder what the difference is in using a native app, vs chrome (or firefox) and excaliraw.

**Comment by thfrei** [comment]
I found one quote from: https://www.pcmag.com/reviews/onyx-boox-note-air-2

> Usable ink speed is largely restricted to Onyx's built-in Notes app, although Onyx has been trying to optimize the experience for other apps. One problem with third-party apps is that the tablet has to cache a bunch of strokes and then send them to the app periodically; this results in weird pauses as a line renders. In OneNote, I could write only slightly less than a line at a time before the Note Air 2 paused for a second to render the strokes. During that interruption, I couldn't see my new strokes on the screen. The app caught up eventually, but the experience is still a little disconcerting. I spoke to Onyx about this issue and I'm confident it is going to fix this lag with a software update that extends the buffer.

Since good "ink" speed is restricted to onyx's native app, and subpar for OneNote (app) and Evernote (app), I guess speed in Chrome would be at best, the same. Maybe we can ask the author, to give it a try, and maybe, they made a software update? :-)

**Reply by pyjoku** [reply]
Well I can tell from my experience:
on my note10+ with pen the experience is great, o I expect any samsung tablet to be without lag
on my onyx Note Air 2 it horrible. I wanted to use  it as note taking in excalidraw and Obsidian. so thats a fail.
What i do is write in the native notes app of the onyx and then paste the note as PDF into my note.
thats fine. and for reading the onyx is great. as I can use the kindle app and other readers and also use my readwise and Pocket app.

Basically it has the features of an ereader, and i would mostly see it as such.
Writing comes on top, but in the native app, Here the experience is the same as in my remarkable.

Also blue and orange backlight, which you can change seperately.
 hope this helps

**Comment by freakedenough** [comment]
**nova air c** isnt any better
https://imgur.com/a/tq1BSKx

**Comment by thfrei** [comment]
Thank you for your report @freakedenough 
During performance investigations, another library came up: https://www.tldraw.com/
Maybe you can check this too? That would be awesome! 🚀

**Comment by freakedenough** [comment]
hello, the website tldraw.com has an identical, if not slightly worse input lag. but this might be caused by the browser, or? its not a native app after all. using einkbro which relies on the integrated webview engine.

heres a video which compares all 3:
onyx native
obsidian
browser

i think its on onyx to sort this out, i also noticed this lag in microsoft whiteboard and onenote.
https://filebin.net/su0l1wp6qndc8i3q/VID_20221025_092605.mp4 
hope this link works, imgur doesnt support more than 60s videos.
might need to replace 3gp with mp4 after downloading...

**Comment by rabfulton** [comment]
Onyx has a sdk, the only way apps get a good writing experience is by integrating with the sdk.

**Reply by Amirroracle** [reply]
One Note seems to work decently, does it integrate directly with the sdk?

**Comment by riekusr** [comment]
I bought a samsung S6 and then an s8 tablet for using excalidraw with it. Applied 'paperfeel screen protector'. 

It works OK there is just 1 bug that is really annoying and that is if you make too many penstrokes too quick, it laggs and loses the last couple of strokes. This happens when you are taking meeting notes and thus is not reliable enough. 

This is a know issue/limitation and the maintainer of obsidian-excalidraw is also aware of this. It is said that excalidraw was not build for that purpose (freehand writing). And I really hope they come with a fix someday. I donate money to obsidian-excalidraw maintainer and am PRO subscriber to have a small impact :) 

EDIT: 
What I am saying, if I think i know the workflow you are looking for, if an beefy S8 tablet can't overpower above issue, no e-tablet wll be able to.

**Comment by pstjohn** [comment]
https://github.com/pstjohn/BooxDraw

in case it's helpful for anyone -- this uses the boox sdk for better latency
